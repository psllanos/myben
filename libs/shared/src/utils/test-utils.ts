import type { AxiosError, AxiosRequestConfig, AxiosResponse } from 'axios'
import type { PlaidError } from 'plaid'

class MockPlaidError extends Error implements AxiosError {
    isAxiosError = true
    code?: string | undefined
    config: AxiosRequestConfig<any> = {}
    request?: any
    response?: AxiosResponse<any, any>
    toJSON = () => ({})

    constructor(error: PlaidError, status: number) {
        super(error.error_message)

        this.response = {
            status,
            statusText: 'ERROR',
            data: error,
            config: {},
            headers: {},
        }
    }
}

export function mockPlaidError(error: PlaidError, status = 400) {
    return new MockPlaidError(error, status)
}

export function axiosSuccess<T>(data: T) {
    return {
        status: 200,
        statusText: '200',
        headers: {},
        config: {},
        data,
    }
}

export function axios400Error<T>(data: T) {
    return {
        config: {},
        response: {
            status: 400,
            statusText: '400',
            headers: {},
            config: {},
            data,
        },
        isAxiosError: true,
    }
}
