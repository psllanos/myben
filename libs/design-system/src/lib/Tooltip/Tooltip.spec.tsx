import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { act } from 'react-dom/test-utils'
import { Tooltip } from '.'

describe('Tooltip', () => {
    describe('when rendered', () => {
        it('should show tooltip content on tooltip trigger `focus`', async () => {
            let component
            await act(async () => {
                component = render(
                    <Tooltip content="Tooltip Content">
                        <button data-testid="tooltip-trigger">Tooltip trigger</button>
                    </Tooltip>
                )
            })

            const tooltipTrrigger = screen.getByTestId('tooltip-trigger')

            tooltipTrrigger.focus()
            await waitFor(() => {
                expect(screen.getByText(/^tooltip content$/i)).toBeVisible()
            })

            expect(component).toMatchSnapshot()
        })
        it('should show tooltip on tooltip trigger `hover`', async () => {
            let component
            await act(async () => {
                component = render(
                    <Tooltip content="Tooltip Content">
                        <button data-testid="tooltip-trigger">Tooltip trigger</button>
                    </Tooltip>
                )
            })

            const tooltipTrrigger = screen.getByTestId('tooltip-trigger')

            userEvent.hover(tooltipTrrigger)
            await waitFor(() => {
                expect(screen.getByText(/^tooltip content$/i)).toBeVisible()
            })

            expect(component).toMatchSnapshot()
        })
    })
})
