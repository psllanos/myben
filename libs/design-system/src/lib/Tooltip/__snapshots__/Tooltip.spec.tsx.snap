// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip when rendered should show tooltip content on tooltip trigger \`focus\` 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        data-testid="tooltip-trigger"
      >
        Tooltip trigger
      </button>
    </div>
    <div
      data-tippy-root=""
      id="tippy-1"
      style="pointer-events: none; z-index: 9999; transition: none; position: absolute; left: 0px; top: 0px; margin: 0px; bottom: 0px; transform: translate(0px, -10px);"
    >
      <div
        class="px-2 py-1 rounded bg-gray-700 border border-gray-600 shadow max-w-[264px] text-sm font-light text-gray-50"
        role="tooltip"
        tabindex="-1"
      >
        Tooltip Content
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      data-testid="tooltip-trigger"
    >
      Tooltip trigger
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tooltip when rendered should show tooltip on tooltip trigger \`hover\` 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        data-testid="tooltip-trigger"
      >
        Tooltip trigger
      </button>
    </div>
    <div
      data-tippy-root=""
      id="tippy-2"
      style="pointer-events: none; z-index: 9999; transition: none; position: absolute; left: 0px; top: 0px; margin: 0px; bottom: 0px; transform: translate(0px, -10px);"
    >
      <div
        class="px-2 py-1 rounded bg-gray-700 border border-gray-600 shadow max-w-[264px] text-sm font-light text-gray-50"
        role="tooltip"
        tabindex="-1"
      >
        Tooltip Content
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      data-testid="tooltip-trigger"
    >
      Tooltip trigger
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
